src/feelpp/benchmarking/reframe/report.py

```diff
@@ -192,68 +256,199 @@ def partialSpeedUp(self):
+                    legendName = df_task["name"].values[0].split('_')[1]
+
+                    fig.add_trace(go.Scatter(x=df_task['num_tasks'], y=df_task['value'],
+                                            mode='lines+markers', name=legendName, line=dict(color=colors[colorIndex])),
+                                            row=1, col=1)
+                    fig.add_trace(go.Scatter(x=df_task['num_tasks'], y=df_task['linearRegression'],
+                                            mode='lines', name=f'{slope:.1f}%', line=dict(dash='dash', color=colors[colorIndex], width=1), visible=False),
+                                            row=1, col=1)
+                    colorIndex += 1
+                    df = pd.concat([df, df_task], ignore_index=True)
+        # Optimality traces
+        fig.add_trace(go.Scatter(x=self.df_partialSpeedup['num_tasks'], y=self.df_partialSpeedup['optimal'], mode='lines', name='Optimal', line=dict(color='grey', width=1), visible=False))
+        fig.add_trace(go.Scatter(x=self.df_partialSpeedup['num_tasks'], y=self.df_partialSpeedup['half optimal'], mode='lines', name='Semi-optimal', line=dict(color='grey', width=1), visible=False))
+        fig.add_trace(go.Scatter(x=self.df_partialSpeedup['num_tasks'], y=self.df_partialSpeedup['optimal'], fill='tonexty', fillcolor='rgba(0, 100, 255, 0.20)', mode='none', name='Optimal area', visible=False))
+        nbCurves = len(fig.data)
+
+        # Table
+        df = df.drop(['optimal', 'half optimal', 'origin'], axis=1)
+        df.loc[:, 'name'] = df['name'].apply(lambda x: x.split('_')[1])
+        table = self.plotTable(df)
+        fig.add_trace(table, row=2, col=1)
+
+        fig.layout.update(title=f'Speed up for {key} phase', updatemenus=self.buildButtons(nbCurves), height=800)
+        return fig
 
-        fig = go.Figure()
-        partialNames = partialPerfNames[key]
-        if key=='solve':
-            partialNames.remove('ksp-niter')
 
-        for t in partialNames:
-            df_task = self.df_partialSpeedup[self.df_partialSpeedup['name'] == t]
-            fig.add_trace(go.Scatter(x=df_task['num_tasks'], y=df_task['value'],
-                                     mode='lines+markers', name=f'{df_task["name"].values[0]}'))
 
-        fig.add_trace(go.Scatter(
-            x=self.df_partialSpeedup['num_tasks'], y=self.df_partialSpeedup['optimal'], mode='lines', name='optimal'))
-        fig.add_trace(go.Scatter(
-            x=self.df_partialSpeedup['num_tasks'], y=self.df_partialSpeedup['half optimal'], mode='lines', name='half optimal'))
-        fig.add_trace(go.Scatter(
-            x=self.df_partialSpeedup['num_tasks'], y=self.df_partialSpeedup['optimal'], fill='tonexty', mode='none', name='optimal'))
+# For debugging purposes with 'your_file.json':
+# Replace 'your_file' with the path to your JSON file
 
-        fig.layout.update(title=f'Speed up for {key} phase')
-        return fig
-    
\ No newline at end of file
+if __name__ == "__main__":
```

Comment:
Same as above. Wrap in `main()` function

Reply:
`debug()`